### PR TITLE
Add a check for configENABLE_MVE to M23, M33 ports

### PR DESCRIPTION
**Pull request by aggarg**
Description
-----------
`configENABLE_MVE` is only applicable to Cortex-M55 and Cortex-M85 ports. It must not be defined to 1 for other ARMv8_m ports.

Test Steps
-----------
Built M23 and M33 example projects locally.

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [ ] I have modified and/or added unit-tests to cover the code changes in this Pull Request.

Related Issue
-----------
https://forums.freertos.org/t/status-of-the-gcc-arm-cm55-and-ntz-ports/19042

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by chinglee-iot — DISMISSED** [review]

File: portable/ARMv8M/non_secure/portable/GCC/ARM_CM23/portmacro.h [hunk 1]
```diff
@@ -60,6 +60,14 @@
 #if ( configTOTAL_MPU_REGIONS == 16 )
     #error 16 MPU regions are not yet supported for this port.
 #endif
+
+#ifndef configENABLE_MVE
+    #define configENABLE_MVE    0
+#else
+    #if( configENABLE_MVE != 0 )
+        #error configENABLE_MVE must be 0 for the Cortex-M23.
```
Comment (RichardBarry) [inline portable/ARMv8M/non_secure/portable/GCC/ARM_CM23/portmacro.h:68]: Must be left undefined, or defined to 0.  Same comment applies in multiple places.

**Review by RichardBarry — CHANGES_REQUESTED** [review]

File: portable/ARMv8M/non_secure/portable/GCC/ARM_CM23/portmacro.h (hunk 1 above)
Comment (aggarg) [reply portable/ARMv8M/non_secure/portable/GCC/ARM_CM23/portmacro.h]: Done.

File: portable/ARMv8M/non_secure/portable/GCC/ARM_CM23/portmacro.h [hunk 2]
```diff
@@ -60,6 +60,14 @@
...
     #error 16 MPU regions are not yet supported for this port.
 #endif
+
+#ifndef configENABLE_MVE
+    #define configENABLE_MVE    0
+#else
+    #if( configENABLE_MVE != 0 )
+        #error configENABLE_MVE must be left undefined, or defined to 0 for the Cortex-M23.
+    #endif
+#endif
```
Comment (Skptak) [inline portable/ARMv8M/non_secure/portable/GCC/ARM_CM23/portmacro.h:70]:
```suggestion
#ifndef configENABLE_MVE
    #define configENABLE_MVE    0
#elif ( configENABLE_MVE != 0 )
        #error configENABLE_MVE must be left undefined, or defined to 0 for the Cortex-M23.
#endif
```

Comment (Skptak) [reply portable/ARMv8M/non_secure/portable/GCC/ARM_CM23/portmacro.h:70]: If this works can we do this instead of the nested `#else/#if`?

Comment (aggarg) [reply portable/ARMv8M/non_secure/portable/GCC/ARM_CM23/portmacro.h]: Done.

**Comment by Skptak** [comment]
I feel this change is inconsistent with https://github.com/FreeRTOS/FreeRTOS-Kernel/pull/966

If we removed the check for the `configTOTAL_MPU_REGIONS` from the CM3_MPU port because it is not possible to configure it, why would we add a check to ensure that `configENABLE_MVE` is disabled in a port that does not support the `MVE` extension? Shouldn't we do the same thing we do in the `ARM_CM3_MPU` port then and allow users to set this, even though it then does nothing?

**Comment by aggarg** [comment]
> Shouldn't we do the same thing we do in the ARM_CM3_MPU port then and allow users to set this, even though it then does nothing?

The reason I removed that config from the M# port is because we are checking a config and that config is used nowhere in the code. That is not the case here as `configENABLE_MVE` is used in the port.c (which is common for all ARMv8M ports).

**Review by Skptak — APPROVED** [review]

**Comment by Skptak** [comment]
> The reason I removed that config from the M3 port is because we are checking a config and that config is used nowhere in the code. That is not the case here as configENABLE_MVE is used in the port.c (which is common for all ARMv8M ports).

Thanks for the clarification! That makes sense

**Review by RichardBarry — APPROVED** [review]

**Pull request merged**